File: Models/ReservaModels.cs
namespace Softawer.Models;

public class Reserva
{
    public int IdReserva { get; set; }
    public int IdUsuario { get; set; }
    public int IdMaquina { get; set; }
    public string NombreMaquina { get; set; } = string.Empty;
    public DateOnly FechaReserva { get; set; }
    public TimeOnly HoraInicio { get; set; }
    public TimeOnly HoraFin { get; set; }
    public string Estado { get; set; } = string.Empty;
    public DateTime FechaCreacion { get; set; }
    public DateTime FechaActualizacion { get; set; }
}

public class CrearReservaRequest
{
    public int IdUsuario { get; set; }
    public int IdMaquina { get; set; }
    public DateOnly FechaReserva { get; set; }
    public TimeOnly HoraInicio { get; set; }
    public TimeOnly HoraFin { get; set; }
}

public class CancelarReservaRequest
{
    public int IdUsuario { get; set; }
}

File: Frontend_Software/src/app/core/models/reserva.models.ts
export type CrearReservaPayload = { idUsuario: number; idMaquina: number; fechaReserva: string; horaInicio: string; horaFin: string };
export type CancelarReservaPayload = { idUsuario: number };
export type Reserva = {
  idReserva: number; idUsuario: number; idMaquina: number; nombreMaquina: string; fechaReserva: string;
  horaInicio: string; horaFin: string; estado: string; fechaCreacion: string; fechaActualizacion: string;
};
